src/morris_sensitivity.jl

```diff
@@ -1,15 +1,23 @@
-function generate_design_matrix(prob::DEProblem,p_range,p_steps,k = 10)
+struct Matrix
+    mat::Array{Array{Float64}}
+    spread::Float64
+end
+
+function generate_design_matrix(prob::DEProblem,p_range,p_steps;k = 10)
     ps = [linspace(p_range[i][1],p_range[i][2],p_steps[i]) for i in 1:length(p_range)]
     indices = [rand(1:i) for i in p_steps]
     all_idxs = Vector{typeof(indices)}(k)
-    all_idxs[1] = copy(indices)
-    i = 2
+    i = 1
     while i <= k
         flag = 0
         indices[rand(1:length(p_range))] += (rand() < 0.5 ? -1 : 1)
-        for j in indices
-            if j > k || j < 1.0
+        for j in 1:length(indices)
+            if indices[j] > p_steps[j] 
```

Comment:
This is making it way too complicated. You only change one index each step. You only need to check if the one index went out of bounds, and if it did, move it 2 in the other direction.